frontend/ui/widget/virtualkeyboard.lua

```diff
@@ -86,6 +87,7 @@ function VirtualKey:init()
             self.keyboard:setKeyboardLayout(next_layout)
         end
         self.hold_callback = function()
+            self.keyboard:emitUtf8ModeKey()
             if util.tableSize(self.key_chars) > 5 then -- 2 or more layouts enabled
```

Comment:
This name "emitUtf8ModeKey()" does not "sound" right - my bad suggestion :)
Here, we're defining the callbacks that happens when tap/hold/swipe on this globe key: so, we're not asking self.keyboard to "emit the utf8 key". At best, we would notify it that "utf8 key emitted or handled or about to be handled".
But still, this sounds strange: why do we need to do that ? Why for this globe key, and why not the others ?! Because your zh keyboard happen to still have this function key (but not "shift" nor "alternate").

So, if we don't want to explain that, may be name it `onSwitchingKeyboardLayout` or something like that, with a comment saying that some "state" keyboards (ie. zh) may want to be notified of that.

Reply:
Agree. Have changed it to this name.

I tried to use it to actually type on my android phone and it turned out the Separation key is hit way more often than the Switch key. So I made some changes to the layout hoping to reduce finger movement. I also changed wording of the Switch key 换字 to 下一字 because it turned out 换 has its traditional form 換. The new wording 下一字 meaning "next character" is also more natural despite being more verbose.